src/main/java/com/aws/greengrass/deployment/DefaultDeploymentTask.java

```diff
@@ -222,29 +219,43 @@ public DeploymentResult call() throws InterruptedException {
 
             // Block this without timeout because it can take a long time for the device to update the config
             // (if it's not in a safe window).
-            DeploymentResult result = deploymentMergeFuture.get();
+            deploymentResult = deploymentMergeFuture.get();
 
             logger.atInfo(DEPLOYMENT_TASK_EVENT_TYPE).setEventType(DEPLOYMENT_TASK_EVENT_TYPE)
                     .log("Finished deployment task");
 
-            componentManager.cleanupStaleVersions();
-            return result;
+            return deploymentResult;
         } catch (PackageLoadingException | DeploymentTaskFailureException | IOException e) {
             logger.atError().setCause(e).log("Error occurred while processing deployment");
-            return new DeploymentResult(DeploymentResult.DeploymentStatus.FAILED_NO_STATE_CHANGE, e);
+            deploymentResult = new DeploymentResult(DeploymentResult.DeploymentStatus.FAILED_NO_STATE_CHANGE, e);
+            return deploymentResult;
         } catch (ExecutionException e) {
             logger.atError().setCause(e).log("Error occurred while processing deployment");
             Throwable t = e.getCause();
             if (t instanceof InterruptedException) {
                 throw (InterruptedException) t;
             } else {
-                return new DeploymentResult(DeploymentResult.DeploymentStatus.FAILED_NO_STATE_CHANGE, t);
+                deploymentResult = new DeploymentResult(DeploymentResult.DeploymentStatus.FAILED_NO_STATE_CHANGE, t);
+                return deploymentResult;
             }
         } catch (InterruptedException e) {
             // DeploymentTask got interrupted while performing a blocking step.
             cancelDeploymentTask(resolveDependenciesFuture, preparePackagesFuture, deploymentMergeFuture);
             // Populate the exception up to the stack
             throw e;
+        } finally {
+            // Clean up stale artifacts from previous deployments after downloading deployment document
+            // if we have desiredPackages then the artifact download has started
+            Map<String, String> failedDeploymentVersions = null;
+            if (desiredPackages != null) {
+                failedDeploymentVersions = desiredPackages.stream()
```

Comment:
I do not agree with the name `failedDeploymentVersions`. This implies that the deployment failed which is not always the case.

Reply:
My understanding here is that cleaning up stale versions will run whenever a deployment is interrupted (we have taken care of every other scenario anyway)
So, if a deployment is interrupted due to
1. deployment revision or cancellation, perhaps `interruptedDeploymentComponentVersions` is a more accurate variable name?
2. device shutdown: i don't believe cleanup happens here but that is okay